### PR TITLE
Improved node descriptions for IntelliSense

### PR DESCRIPTION
**Pull request by vovamarch**

### CODE REVIEW

File: tools/projmgr/schemas/common.schema.json [hunk 1]
```diff
@@ -331,14 +363,14 @@
...
-        "C-CPP":        { "$ref": "#/definitions/ArrayOfStrings", "description": "List of C & CPP flags" },
-        "ASM":          { "$ref": "#/definitions/ArrayOfStrings", "description": "List of ASM flags"  },
-        "Link":         { "$ref": "#/definitions/ArrayOfStrings", "description": "List of Linker flags" },
-        "Link-C":       { "$ref": "#/definitions/ArrayOfStrings", "description": "List of Linker flags for pure C project" },
-        "Link-CPP":     { "$ref": "#/definitions/ArrayOfStrings", "description": "List of Linker flags for project with C++ files" },
-        "Library":      { "$ref": "#/definitions/ArrayOfStrings", "description": "List of Linker flags for libraries handling" },
+        "C":            { "$ref": "#/definitions/ArrayOfStrings", "description": "Flags for *.c files" },
+        "CPP":          { "$ref": "#/definitions/ArrayOfStrings", "description": "Flags for *.cpp files" },
+        "C-CPP":        { "$ref": "#/definitions/ArrayOfStrings", "description": "Flags for *.c  and *.cpp files" },
+        "ASM":          { "$ref": "#/definitions/ArrayOfStrings", "description": "Flags for assember source files"  },
```
Comment (brondani) [inline tools/projmgr/schemas/common.schema.json:369]: Typo:
```suggestion
        "ASM":          { "$ref": "#/definitions/ArrayOfStrings", "description": "Flags for assembler source files"  },
```

File: tools/projmgr/schemas/common.schema.json [hunk 2]
```diff
@@ -445,35 +477,35 @@
...
     },
     "BuildProcessorFpu": {
       "enum": [ "on", "dp", "sp", "off" ],
-      "description": "Controls usage of FPU registers (S-Registers that are used for Helium and FPU hardware): on (enabled) | dp (double precision) | sp (single precision) | off (disabled)"
+      "description": "Controls usage of FPU instructions:\non (enabled)\n dp (double precision)\n sp (single precision)\n off (disabled) ."
     },
     "ProcessorDsp": {
       "enum": [ "on", "off" ],
-      "description": "Controls usage of DSP: on (enabled) | off (disabled)"
+      "description": "Controls usage of SIMD instructions:\n on (enabled) | off (disabled) ."
```
Comment (brondani) [inline tools/projmgr/schemas/common.schema.json:492]: The `dsp` node concerns DSP extensions, for example disabling the DSP for Cortex-M33 will result in the following AC6 flag:  [`-mcpu=Cortex-M33+nodsp`](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/buildmgr/cbuildgen/config/AC6.6.16.2.cmake#L365C25-L365C47)
SIMD flags such as [`-fvectorize`](https://developer.arm.com/documentation/dui0774/l/Compiler-Command-line-Options/-fvectorize---fno-vectorize) are not yet abstracted via dedicated yaml nodes.
```suggestion
      "description": "Controls usage of DSP extensions:\n on (enabled) | off (disabled) ."
```

File: tools/projmgr/schemas/cproject.schema.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
   "$id": "https://raw.githubusercontent.com/Open-CMSIS-Pack/devtools/schemas/projmgr/2.6.0/tools/projmgr/schemas/cproject.schema.json",
-  "title": "CMSIS cproject",
-  "description": "source files and pre-configured components of a sub-project that builds independently",
+  "title": "CMSIS-Toolbox Project\nDocumentation: https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#project",
```
Comment (brondani) [inline tools/projmgr/schemas/cproject.schema.json:4]:
```suggestion
  "title": "CMSIS Project\nDocumentation: https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#project",
```

**Review by brondani — COMMENTED** [review]
I would also suggest revising the final dot `.` in all descriptions and make it consistent.

File: tools/projmgr/schemas/csolution.schema.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
   "$id": "https://raw.githubusercontent.com/Open-CMSIS-Pack/devtools/schemas/projmgr/2.6.0/tools/projmgr/schemas/csolution.schema.json",
-  "title": "CMSIS csolution",
-  "description": "complete scope of an application composed of sub-projects; defines a target and build types",
+  "title": "CMSIS-Toolbox Solution\nDocumentation: https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#solution",
```
Comment (brondani) [inline tools/projmgr/schemas/csolution.schema.json:4]:
```suggestion
  "title": "CMSIS Solution\nDocumentation: https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#solution",
```

File: tools/projmgr/schemas/common.schema.json (hunk 1 above)
Comment (vovamarch) [reply tools/projmgr/schemas/common.schema.json]: Addressed

File: tools/projmgr/schemas/cproject.schema.json (hunk 3 above)
Comment (vovamarch) [reply tools/projmgr/schemas/cproject.schema.json]: Addressed

File: tools/projmgr/schemas/csolution.schema.json (hunk 4 above)
Comment (vovamarch) [reply tools/projmgr/schemas/csolution.schema.json]: Addressed

File: tools/projmgr/schemas/common.schema.json (hunk 2 above)
Comment (jkrech) [reply tools/projmgr/schemas/common.schema.json:492]: How about: "Controls usage of DSP instruction set extensions by the compiler as supported by the processor"

Comment (vovamarch) [reply tools/projmgr/schemas/common.schema.json]: Updated as suggested, but without "by the compiler" as we don't specify that in other features either.

**Review by jkrech — APPROVED** [review]
LGTM

**Review by brondani — APPROVED** [review]

**Pull request merged**